vscode-dotnet-runtime-library/src/Acquisition/AcquisitionInvoker.ts

```diff
@@ -137,33 +138,45 @@ You will need to restart VS Code after these changes. If PowerShell is still not
     {
         let knownError = false;
         let error = null;
-        let command = '';
+        let command = null;
+
+        const possibleCommands =
+        [
+            CommandExecutor.makeCommand(`powershell.exe`, []),
+            CommandExecutor.makeCommand(`%SystemRoot%\\System32\\WindowsPowerShell\\v1.0\\powershell.exe`, []),
+            CommandExecutor.makeCommand(`pwsh`, []),
+            CommandExecutor.makeCommand(`powershell`, []),
+            CommandExecutor.makeCommand(`pwsh.exe`, [])
+        ];
 
         try
         {
             // Check if PowerShell exists and is on the path.
-            const commandWorking = await new CommandExecutor(this.eventStream, this.utilityContext).TryFindWorkingCommand([`powershell.exe`,
-                `%SystemRoot%\\System32\\WindowsPowerShell\\v1.0\\powershell.exe`, `pwsh`, `powershell`, `pwsh.exe`]);
-            if(!commandWorking[1])
+            const workingCommandIndex = await new CommandExecutor(this.eventStream, this.utilityContext).tryFindWorkingCommand(possibleCommands);
+
+            if(workingCommandIndex === -1)
             {
                 knownError = true;
                 const err = Error(this.noPowershellError);
                 error = err;
             }
             else
             {
-                command = commandWorking[0];
+                command = possibleCommands[workingCommandIndex];
             }

```

Comment:
nit: instead of returning indexes, why not return the found command to prevent any change of array indexing errors?

Reply:
Good point, that makes it much cleaner. The way it was initially written didnt allow for this but now it can be done.